packages/toi/index.ts

```diff
@@ -647,69 +647,115 @@ export namespace obj {
     );
 
   /**
-   * Checks that the object obeys a strict structure of properties and property-level validators.
+   * Checks that the object obeys a strict structure of properties and property-level validators. The structure needs to define all properties in a plain JavaScript object and its prototype chain is not inspected.
    *
    * It is a strict 1:1 validation. Keys not found on the value will trigger a {@link ValidationError},
-   * including keys found in the `value` but not in the structure.
+   * including keys found in the `value` but not in the structure. This excludes non-enumerable properties (like `toString` or other built-in functions) or properties defined on symbols.
+   * You can disable this strict 1:1 mapping by setting the `lenient` option to `true.
    *
-   * The `value` is never returned as-is, but is made a copy of with the own properties according to
+   * The prototype chain of the passed value will be walked and inherited properties will be validated. You can disable this by setting the `own` option to `true`.
+   *
+   * The `value` is never returned as-is, but is made a copy of with the properties according to
    * the defined `structure`. Therefore, the property-level validators may transform the property
    * values. This safeguards from `instanceof` checks, including prototype-hijacking and similar
-   * issues.
+   * issues. The returned value will always have the Object prototype, as the prototype chain is not copied.
    *
    * @param structure the definition structure of the object
-   * @param options the validation options, like optional (missing) fields
+   * @param options the validation options
    */
-  export const keys = <X extends object, Y>(
+  export const keys = <X extends object, Y, M extends Y>(
     structure: { [K in keyof Y]: Validator<any, Y[K]> },
-    options?: { missing?: (keyof Y)[] }
+    options?: {
+      /** Set the keys that may be missing in the value. */
+      missing?: (keyof M)[];
+      /** Only validate the value's own properties and ignore inherited properties. */
+      own?: boolean;
+      /** Don't validate that value matches 1:1 with the structure. */
+      lenient?: boolean;
+    }
```

Comment:
You can make the `options` param have a default value set to `{}`. That way, you won't have to do `!options` everywhere.